app/models/quarter.rb

```diff
@@ -1,24 +1,50 @@
 class Quarter
+  include Comparable
+  PARSE_REGEX = /\AQ(\d)-(\d{4})\z/
+
+  def self.for_date(date)
+    new(date: date)
+  end
+
   def self.current
     new(date: Date.current)
   end
 
+  def self.parse(string)
+    match = string.match(PARSE_REGEX)
+    raise ArgumentError, "String to parse as Quarter must match QX-YYYY format" unless match
+    number = Integer(match[1])
+    year = Integer(match[2])
+    quarter_month = ((number - 1) * 3) + 1
+    date = Date.new(year, quarter_month).beginning_of_month
+    new(date: date)
+  end
+
   attr_reader :date
   attr_reader :number
+  attr_reader :to_s
   attr_reader :year
+  delegate :beginning_of_quarter, :end_of_quarter, to: :date
 
   def initialize(date:)
     @date = date.freeze
     @year = date.year.freeze
     @number = QuarterHelper.quarter(date).freeze
+    @to_s = "Q#{number}-#{year}".freeze
   end
 
   def next_quarter
-    self.class.new(date: date.advance(months: 3))
+    advance(months: 3)
   end
 
+  alias succ next_quarter
```

Comment:
Instead of aliasing this, should we just find/replace with `succ` and `previous`?

Reply:
We need the `succ` method so that Quarters work in ranges...and we provide `next_quarter` to have a better named version of it. =)